src/executables/master_server_monitor.cpp

```diff
@@ -49,65 +48,322 @@ server_monitor::server_monitor ()
+	   m_server_entry_map->size ());
+
+}
+
+void
+server_monitor::revive_server_with_name (std::string server_name)
+{
+  int error_code;
+  const int SERVER_MONITOR_UNACCEPTABLE_REVIVE_TIMEDIFF_IN_MSECS = prm_get_integer_value (
+	      PRM_ID_HA_UNACCEPTABLE_PROC_RESTART_TIMEDIFF_IN_MSECS);
+  std::chrono::steady_clock::time_point tv;
+  int out_pid = 0;
+
+  auto entry = m_server_entry_map->find (server_name);
+  if (entry != m_server_entry_map->end ())
+    {
+      entry->second.set_need_revive (true);
+      tv = std::chrono::steady_clock::now ();
+      auto timediff = std::chrono::duration_cast<std::chrono::milliseconds> (tv -
+		      entry->second.get_last_revive_time()).count();
+      if (timediff > SERVER_MONITOR_UNACCEPTABLE_REVIVE_TIMEDIFF_IN_MSECS)
+	{
+	  entry->second.set_last_revive_time (std::chrono::steady_clock::now ());
+	  server_monitor_try_revive_server (entry->second.get_exec_path(), entry->second.get_argv(), &out_pid);
+	  assert (out_pid > 0);
+	  entry->second.set_pid (out_pid);
+	  master_Server_monitor->server_monitor_produce_job (SERVER_MONITOR_CONFIRM_REVIVE_ENTRY, -1, "", "",
+	      entry->first);
+	  return;
+	}
+      else
+	{
+	  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : Process failure repeated within a short period of time. pid : %d\n",
+		   entry->second.get_pid());
+	  m_server_entry_map->erase (entry);
+	  return;
+	}
+    }
+}
+
+void
+server_monitor::server_monitor_check_server_revived (std::string server_name)
+{
+  int error_code;
+  auto entry = m_server_entry_map->find (server_name);
+  if (entry != m_server_entry_map->end ())
+    {
+      error_code = kill (entry->second.get_pid (), 0);
+      if (error_code == ESRCH)
+	{
+	  master_Server_monitor->server_monitor_produce_job (SERVER_MONITOR_REVIVE_ENTRY, -1, "", "", entry->first);
+	}
+      else if (entry->second.get_need_revive ())
+	{
+	  std::this_thread::sleep_for (std::chrono::milliseconds (1000));
+	  master_Server_monitor->server_monitor_produce_job (SERVER_MONITOR_CONFIRM_REVIVE_ENTRY, -1, "", "",
+	      entry->first);
+	}
+      else
+	{
+	  fprintf (stdout, "[SERVER_REVIVE_DEBUG] : Server revive success. pid : %d\n", entry->second.get_pid());
+	}
+      return;
+    }
+  master_Server_monitor->server_monitor_produce_job (SERVER_MONITOR_REVIVE_ENTRY, -1, "", "", server_name);
+}
+
+void
+server_monitor::server_monitor_produce_job (server_monitor_job_type job_type, int pid, std::string exec_path,
+    std::string args, std::string server_name)
+{
+  server_monitor::server_monitor_job job (job_type, pid, exec_path, args, server_name);
+  std::unique_lock<std::mutex> lock (m_monitor_mutex_full);
+
+  constexpr int period_in_secs = 1;
+  while (!m_monitor_cv_full.wait_for (lock, std::chrono::seconds (period_in_secs), [this] ()
+  {
+    return !m_job_queue->is_full ();
+    }));
+  m_job_queue->produce (job);
+  m_monitor_cv_full.notify_all();
```

Comment:
```suggestion
  server_monitor::server_monitor_job job (job_type, pid, exec_path, args, server_name);
  std::unique_lock<std::mutex> lock (m_monitor_mutex_producer);

  constexpr int period_in_secs = 1;
  while (!m_monitor_cv_producer.wait_for (lock, std::chrono::seconds (period_in_secs), [this] ()
  {
    return !m_job_queue->is_full ();
    }));
  m_job_queue->produce (job);
  m_monitor_cv_consumer.notify_all();
```

1. 명확한 의미를 갖도록 변수들 이름 변경
2. consumer에게 notify를 보내야합니다.

Reply:
수정하였습니다.